api/src/test/java/org/openmrs/module/fhir2/providers/r3/PatientFhirR3ResourceProviderTest.java

```diff
@@ -415,4 +423,70 @@ public void getPatientHistoryByWithWrongId_shouldThrowResourceNotFoundException(
 		assertThat(patientFhirResourceProvider.getPatientHistoryById(idType).isEmpty(), is(true));
 		assertThat(patientFhirResourceProvider.getPatientHistoryById(idType).size(), equalTo(0));
 	}
+	
+	@Test
+	public void createPatient_shouldCreateNewPatient() {
+		when(patientService.create(any(org.hl7.fhir.r4.model.Patient.class))).thenReturn(patient);
+
+		MethodOutcome result = patientFhirResourceProvider.createPatient(Patient30_40.convertPatient(patient));
+		assertThat(result, notNullValue());
+		assertThat(result.getCreated(), is(true));
+		assertThat(result.getResource(), equalTo(patient));
+	}
+
+	@Test
+	public void deletePatient_shouldDeleteRequestedPatient() {
+
+		when(patientService.delete(PATIENT_UUID)).thenReturn(patient);
+
+		OperationOutcome result = patientFhirResourceProvider.deletePatient(new IdType().setValue(PATIENT_UUID));
+		assertThat(result, notNullValue());
+		assertThat(result.getIssue(), notNullValue());
+		assertThat(result.getIssueFirstRep().getSeverity(), equalTo(OperationOutcome.IssueSeverity.INFORMATION));
+		assertThat(result.getIssueFirstRep().getDetails().getCodingFirstRep().getCode(), equalTo("MSG_DELETED"));
+		assertThat(result.getIssueFirstRep().getDetails().getCodingFirstRep().getDisplay(),
+		    equalTo("This resource has been deleted"));
+	}
+
+	@Test(expected = ResourceNotFoundException.class)
+	public void deletePatient_shouldThrowResourceNotFoundException() {
+		IdType id = new IdType();
+		id.setValue(WRONG_PATIENT_UUID);
+		OperationOutcome patient = patientFhirResourceProvider.deletePatient(id);
+		assertThat(patient, nullValue());
+	}
```

Comment:
We don't need an assertion when we have an expected exception.

```suggestion
		id.setValue(WRONG_PATIENT_UUID);
		
		OperationOutcome patient = patientFhirResourceProvider.deletePatient(id);
	}
```